src/qtbase.mk

```diff
@@ -9,7 +9,7 @@ $(PKG)_CHECKSUM := 4ef921c0f208a1624439801da8b3f4344a3793b660ce1095f2b7f5c4246b9
 $(PKG)_SUBDIR   := $(PKG)-everywhere-src-$($(PKG)_VERSION)
 $(PKG)_FILE     := $(PKG)-everywhere-src-$($(PKG)_VERSION).tar.xz
 $(PKG)_URL      := https://download.qt.io/official_releases/qt/5.14/$($(PKG)_VERSION)/submodules/$($(PKG)_FILE)
-$(PKG)_DEPS     := cc dbus fontconfig freetds freetype harfbuzz jpeg libmysqlclient libpng openssl pcre2 postgresql sqlite zlib zstd
+$(PKG)_DEPS     := cc dbus fontconfig freetds freetype harfbuzz jpeg libmysqlclient libpng openssl pcre2 postgresql sqlite zlib zstd jasper libmng tiff libwebp lcms lzma
```

Comment:
Aren't these extra deps introduced by qtimageformats rather than qtbase?

Reply:
Hmm. Indeed, looks like these deps introduced by separate Qt internal plugin `Qt Image Formats`, which in turn provided by `qtimageformats` package in MXE. But if so, then my app build should fail due to missing plugin itself. I did neither install this package, nor Qt application being build has dependency on corresponding plugin. Only `qtbase`. And only after I installed these extra packages it was build successfully. Otherwise o failed at link stage with `cannot find -l<...>` message for each extra dep. That's why I decided they correspond to qtbase.
I guess we discovered another Qt bug(feature?). Libraries list for linker is being generated by mechanism implemented in qtbase which tracks dependencies for all qt addon modules and plugins not accounting whether they are actually used in specific app build they generated for, i.e. module just "reports" dependencies of all plugins it uses, and although app isn't actually link with them at runtime and may be built, these libs are stupidly passed to linker which complains, because cannot find them. But that's just a guess, because otherwise I don't have any other explanation. Need to check...

Reply:
Ok, I've got the point. It turned out to be a documented feature rather than bug and, of course, it's more general than this specific `QtCore -> Qt Image Formats` dependency relation. [Qt Cmake documentations](https://doc.qt.io/qt-5/qtcore-cmake-qt5-import-plugins.html) describe it:

> ...
> Qt provides plugin types such as **imageformats**, platforms, and sqldrivers.
> If `qt5_import_plugins()` isn't called, the target **automatically links against a sane set of default plugins**, for each Qt module that the target is linked against.
> ...
> Note: This macro imports plugins from static Qt builds only. On shared builds, it does nothing.
> ...

Actually this behavior isn't much different from what I guessed above, because it doesn't link against plugin itself but does against its dependencies, so I would consider this a bug. But the behavior itself imposes a problem into MXE, regardless of this bug being fixed or not, because now MXE Qt packages dependencies needs to be rewritten in a way I can't imagine of - for each MXE package containing such a default plugin (where decision of being default or not, depends on Qt internal logic, possibly version-specific) there is bidirectional dependency between it and a package containing corresponding module (such as `qtbase`). Maybe it makes sense to find out how similar problem being solved in other packaging systems such as Linux distributions...

Reply:
> If qt5_import_plugins() isn't called, the target automatically links against a sane set of default plugins

I'd say we should set this explicitly (maybe in the toolchain file includes) and investigate how subsequent calls in project files interact with that setting.

Reply:
> I'd say we should set this explicitly (maybe in the toolchain file includes)

Ignore that, it's an executable-specific setting.

A [simple cmake test](https://github.com/ArtemPisarenko/mxe/pull/1) builds and runs fine without the extra image deps or setting `qt5_import_plugins`.

Reply:
Yes, it turned out to be I've got things wrong again :) Seems like commit c943c9b changes Qt behavior and, indeed, these changes are inapropriate now.

I've tested all cominations of build/run Qt applications:
- app doesn't depend on any non-default Qt plugin (similar to simple test you added);
- app using extra Webp image format, having `qt5_import_plugins(mxe_qt_app INCLUDE_BY_TYPE imageformats Qt5::QWebpPlugin)` in cmake file and calling `QImage image(100, 100, QImage::Format_RGB32); Q_ASSERT(image.save("image.webp"));`;
- linking against static Qt;
- linking against shared Qt;
- building/deploying from MXE environment which has only `qtbase` package installed;
- building/deploying from MXE environment which has `qtbase` and `qtimageformats` packages installed.

App is being built and run successfully in all combinations above and its second variant doesn't fail at Q_ASSERT in runtime in combinations where it being built/deployed from MXE with `qtimageformats` installed. In static variants Qt cmake configuration is smart enough to import only those plugins it found itself.

I personally like this behavior. Although, correct deploying app with plugins isn't simple, especially for beginers, because app doesn't report any diagnostic info why it failed (although maybe I just don't know how to debug it). I mean that it isn't enough to collect DLLs dependencies from app .exe only and copy necessary plugins, but also each plugin DLL dependencies must be collected too, e.g. in my test I have to do `mkdir deploy && (x86_64-w64-mingw32.shared-peldd mxe_qt_app.exe /mxe/usr/x86_64-w64-mingw32.shared/qt5/plugins/imageformats/qwebp.dll -a -w d3d11.dll -w dxgi.dll | xargs cp -t deploy/) && cp -r /mxe/usr/x86_64-w64-mingw32.shared/qt5/plugins/platforms /mxe/usr/x86_64-w64-mingw32.shared/qt5/plugins/imageformats deploy/`